src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/logger.html.twig

```diff
@@ -33,103 +33,136 @@
-                    </tr>
-                {% endif %}
-            {% else %}
-                <tr><td colspan="4"><em>No logs available for {{ priority }} priority.</em></td></tr>
-            {% endfor %}
-        </table>
     {% else %}
-        <p>
-            <em>No logs available.</em>
-        </p>
+
+        {# sort collected logs in groups #}
+        {% set deprecation_logs, debug_logs, info_and_error_logs = [], [], [] %}
+        {% for log in collector.logs %}
+            {% if log.context.level is defined and log.context.type is defined and log.context.type in [constant('E_DEPRECATED'), constant('E_USER_DEPRECATED')] %}
+                {% set deprecation_logs = deprecation_logs|merge([log]) %}
+            {% elseif log.priorityName == 'DEBUG' %}
+                {% set debug_logs = debug_logs|merge([log]) %}
+            {% else %}
+                {% set info_and_error_logs = info_and_error_logs|merge([log]) %}
+            {% endif %}
+        {% endfor %}
+
+        <div class="sf-tabs">
+            <div class="tab">
+                <h3 class="tab-title">Info. &amp; Errors <span class="badge">{{ info_and_error_logs|length }}</span></h3>
+
+                <div class="tab-content">
+                    {% if info_and_error_logs is empty %}
+                        <div class="empty">
+                            <p>There are no log messages of this level.</p>
+                        </div>
+                    {% else %}
+                        {{ helper.render_table(info_and_error_logs, true) }}
+                    {% endif %}
+                </div>
+            </div>
+
+            <div class="tab">
+                <h3 class="tab-title">Deprecations <span class="badge">{{ deprecation_logs|length }}</span></h3>
+
+                <div class="tab-content">
+                    {% if deprecation_logs is empty %}
+                        <div class="empty">
+                            <p>There are no log messages about deprecated features.</p>
+                        </div>
+                    {% else %}
+                        {{ helper.render_table(deprecation_logs, false, true) }}
+                    {% endif %}
+                </div>
+            </div>
+
+            <div class="tab">
+                <h3 class="tab-title">Debug <span class="badge">{{ debug_logs|length }}</span></h3>
+
+                <div class="tab-content">
+                    {% if debug_logs is empty %}
+                        <div class="empty">
+                            <p>There are no log messages of this level.</p>
+                        </div>
+                    {% else %}
+                        {{ helper.render_table(debug_logs) }}
+                    {% endif %}
+                </div>
+            </div>
+        </div>
+
     {% endif %}
 {% endblock %}
 
+{% macro render_table(logs, show_level = false, is_deprecation = false) %}
+    {% import _self as helper %}
```

Comment:
I've made all the changes suggested in your other comments, but I'm struggling with this one.

First I don't really know why `{% from ... %}` is recommended instead of `{% import ...`. Second, I though that perfixing all utility macros with the `helper.` prefix would be better for code legibility. But of course this could be a bad idea :)  I just need some more comments about this. Thanks.